src/node.cc

```diff
@@ -4712,15 +4736,39 @@ inline int Start(Isolate* isolate, void* isolate_context,
 
 #if ENABLE_TTD_NODE
   // Start time travel after environment is loaded
-  if (s_doTTRecord) {
-    fprintf(stderr, "Recording started (after main module loaded)...\n");
+  if (s_doTTRecord || s_doTTReplay) {
+    if (s_doTTRecord) {
+      fprintf(stderr, "Recording started (after main module loaded)...\n");
+    }
+
     JsTTDStart();
   }
 #endif
 
   env.set_trace_sync_io(trace_sync_io);
 
+#if ENABLE_TTD_NODE
+  if (s_doTTReplay) {
+    int64_t nextEventTime = -2;
+    bool continueReplayActions = true;
+
+    while (continueReplayActions) {
+      continueReplayActions = v8::Isolate::RunSingleStepOfReverseMoveLoop(
+          isolate,
+          &s_ttdStartupMode,
+          &nextEventTime);
+
+      // don't continue replay actions if we are not in debug mode
+      continueReplayActions &= s_doTTEnableDebug;
+    }
+
+    // We are done just dump the process.
+    // In the future we might want to clean up more.
+    exit(0);
+  } else {
```

Comment:
Minor nit: instead of

```cpp
} else {
#else
{
#endif
```

you could do

```cpp
} else
#endif
{
```

Although I suppose the linter might get mad at you for that...

Reply:
src\node.cc:4768:  If an else has a brace on one side, it should have it on both  [readability/braces] [5]
src\node.cc:4768:  If/else bodies with multiple statements require braces  [readability/braces] [4]